fastlane/Fastfile

```diff
@@ -278,6 +281,10 @@ end
   # bundle exec fastlane build_and_upload_app_store create_github_release:true
   #############################################################################
   lane :build_and_upload_app_store do | options |
+    configure_apply()
+    sh("rake dependencies:pod:clean")
+    cocoapods()
```

Comment:
I ported this from the lane that builds the beta version.

Originally, I had CI running `configure_apply`. This is a better approach because it moves more of the imperative set of steps to a CI-agnostic location.

Because CI already runs `pod install`, I wondered whether to keep the two instructions related to it. https://github.com/Automattic/simplenote-macos/blob/97fd3ea29ec300d00fffe1b84b5786aec7977c24/.circleci/config.yml#L17-L19

I said just before that I value a CI-agnostic setup, but in the case of external dependencies it's actually valuable to explicitly install them in CI, so we can tap into its caching mechanism.

What do you think? If we decide to remove these, I'll do it in a dedicated PR to cover both production and beta lanes.